CMakeLists.txt

```diff
@@ -18,16 +17,6 @@ if (NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/CMakeCache.txt)
   endif ()
 endif ()
 
-if (PROJECT_IS_TOP_LEVEL AND USE_CONAN)
-  include(cmake/conan_cmake.cmake)
-
-  if (BUILD_TESTING OR BUILD_EXAMPLES)
-    set(CONAN_OPTIONS "with_test_deps=True")
-  endif ()
-
-  install_packages_via_conan("${CMAKE_SOURCE_DIR}/conanfile.py" "${CONAN_OPTIONS};boost/*:header_only=True")
```

Comment:
Now that I see it here: is there a way to inject arguments to the conan provider? Probably not, right?

Reason: we don't need the non-header-only parts of boost and makes CI faster.
Specifying it in the conanfile itself sets it for all dependents (for some reason..), which is not what we want